### DISCUSSION TITLE
Cannot upload files to the root directory using Sardine library

### DISCUSSION THREAD

**Discussion by P529** [lookfirst/sardine#390] [Q&A]
I am using the Sardine library in Java to upload files to a WebDAV server. When attempting to upload files to the root directory (.../remote.php/dav/files/p529/), I encounter the following error:

`java.lang.RuntimeException: org.apache.http.client.ClientProtocolException
...
org.apache.http.client.NonRepeatableRequestException: Cannot retry request with a non-repeatable request entity.
`

The issue arises only when trying to upload to the root directory; creating folders in the root directory and uploading to subdirectories works fine.

**Steps to reproduce in the application im running (ServerBackup Minecraft Mod):**

    1. Use Sardine library to upload a file to the root directory of the specified WebDAV server.
    2. Observe the exception mentioned above.
    
**Expected behavior:**
I expect to be able to upload files to the root directory using the Sardine library without encountering the NonRepeatableRequestException.

**Additional information:**

1. Sardine Library Version: 5.10
2. Java Version: 8.372 Azul

This is the Code Snippet where I encounter the issue:

[GitHub Gist of usage](https://gist.github.com/P529/4901fdb2e20ae7e05796d9a7d43c83aa)

This is the Log I receive in my application, I've tried to achieve more verbose logging but was not able to.

[ErrorLog.txt](https://github.com/lookfirst/sardine/files/11521107/ErrorLog.txt)

I've also tried to use sardine.put() in combination with a ByteArray and that worked, however I have to deal with rather large files and relatively little ram so its not an option.

Maybe this information also helps, im using nextCloud and their WebDav functionality and checked with WinScp if I am able to upload files directly to the root directory which also works.

Best regards P529

**Comment by pavelsalamon** [comment]
Try using pre-emptive auth, it will work as long as Basic auth is used. See comments in https://github.com/lookfirst/sardine/issues/103 for more info.
